modules/dns_nsupdate/dns_nsupdate_gss_main.rb

```diff
@@ -6,17 +6,12 @@ class Record < Proxy::Dns::Nsupdate::Record
     include Proxy::Kerberos
     attr_reader :tsig_keytab, :tsig_principal
 
-    def initialize options = {}
+    def initialize(server = nil, ttl = nil)
       @tsig_keytab = ::Proxy::Dns::NsupdateGSS::Plugin.settings.dns_tsig_keytab
       @tsig_principal = ::Proxy::Dns::NsupdateGSS::Plugin.settings.dns_tsig_principal
-      raise "Keytab not configured via dns_tsig_keytab for DNS GSS-TSIG support" unless tsig_keytab
-      raise "Unable to read dns_tsig_keytab file at #{tsig_keytab}" unless File.exist?(tsig_keytab)
-      raise "Kerberos principal required - check dns_tsig_principal setting" unless tsig_principal
```

Comment:
I'm not sure we should filter them out when there's no value, it could be unexpected, I think it's preferable to follow what the user entered but to raise an error.

A likely scenario is even in our own installer or config management systems - if a setting isn't initialised or set properly, a config file could be templated with a blank setting, e.g. `:foo: <%= @uninitialised_variable %>`.  It might be safer to let the proxy initialise the setting to nil and throw an error saying that the :foo setting's nil or not a file.